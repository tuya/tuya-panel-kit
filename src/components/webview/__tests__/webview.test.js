import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import WebView from '../index';

jest.mock('WebView', () => 'WebView');

describe('WebView Component', () => {
  it('basic render', () => {
    const component = renderer
      .create(<WebView source="https://www.tuya.com" title="Tuya" />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  // it('should get and set ref correctly', () => {
  //   const component = renderer.create(<WebView source="https://www.tuya.com" />);
  //   const instance = component.getInstance();
  //   instance.setInstance('aaa');
  //   expect(instance.__webview).toBe('aaa');
  //   expect(instance.getInstance()).toBe('aaa');
  // });
});
