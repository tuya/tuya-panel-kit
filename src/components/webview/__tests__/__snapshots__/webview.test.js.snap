// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebView Component basic render 1`] = `
<WebView
  automaticallyAdjustContentInsets={true}
  decelerationRate="normal"
  domStorageEnabled={true}
  javaScriptEnabled={true}
  scalesPageToFit={true}
  source={
    Object {
      "uri": "https://www.tuya.com",
    }
  }
  startInLoadingState={true}
  style={
    Object {
      "flex": 1,
      "height": 1246,
    }
  }
/>
`;
